src/components/ArticleList.jsx

```diff
@@ -1,22 +1,27 @@
-import {getArticles} from "../api"
+import {getArticles, getArticlesByTopic} from "../api"
 import ArticleCard from "./ArticleCard"
 import { useEffect, useState } from "react"
 import Loading from "./Loading"
 /* eslint-disable react/prop-types */
 
-function ArticleList({searchTerm}){
+function ArticleList({searchTerm, topic}){
     const [articles, setArticles] = useState([])
     const [loading, setLoading] = useState(true)
 
     useEffect(() => {
         setLoading(true)
-        getArticles(searchTerm).then((articles) => {
-            setArticles(articles)
+
+        const fetchArticles = topic ? getArticlesByTopic : getArticles
```

Comment:
is there a way we could use the same function to do both? fetch by topic when there is one and fetch all when there isnt?